zebra-scan/src/storage/db/tests.rs

```diff
@@ -1,3 +1,58 @@
 //! General scanner database tests.
 
+use std::sync::Arc;
+
+use zebra_chain::{
+    block::{Block, Height},
+    parameters::Network::{self, *},
+    serialization::ZcashDeserializeInto,
+};
+use zebra_state::TransactionIndex;
+
+use crate::{
+    storage::Storage,
+    tests::{FAKE_SAPLING_VIEWING_KEY, ZECPAGES_SAPLING_VIEWING_KEY},
+    Config,
+};
+
+#[cfg(test)]
 mod snapshot;
+
+/// Returns an empty `Storage` suitable for testing.
+pub fn new_test_storage(network: Network) -> Storage {
+    Storage::new(&Config::ephemeral(), network)
+}
+
+/// Add fake keys to `storage` for testing purposes.
+pub fn add_fake_keys(storage: &mut Storage) {
+    // Snapshot a birthday that is automatically set to activation height
+    storage.add_sapling_key(&ZECPAGES_SAPLING_VIEWING_KEY.to_string(), None);
+    // Snapshot a birthday above activation height
+    storage.add_sapling_key(&FAKE_SAPLING_VIEWING_KEY.to_string(), Height(1_000_000));
+}
+
+/// Add fake results to `storage` for testing purposes.
+pub fn add_fake_results(storage: &mut Storage, network: Network, height: Height) {
+    let blocks = match network {
+        Mainnet => &*zebra_test::vectors::CONTINUOUS_MAINNET_BLOCKS,
+        Testnet => &*zebra_test::vectors::CONTINUOUS_TESTNET_BLOCKS,
+    };
+
+    let block: Arc<Block> = blocks
+        .get(&height.0)
+        .expect("block height has test data")
```

Comment:
Since `blocks` has a constant length, currently set to 11, and `height` is a parameter the caller sets arbitrarily, it's easy to trigger a panic here.

I personally still struggle with formulating error messages from time to time. The message here says "block height has test data", which is the opposite of what's happening when displayed to the user. On the other hand, the message explains to the developer why we can expect an `Ok` here. Do you recommend using this style more often than the other?

Reply:
I think usual Rust style is "unexpected (bad thing): (good thing) must happen". That helps remove the ambiguity.